### ISSUE TITLE
-Wnoexcept warning with GCC 9.1.0

### ISSUE THREAD

**Issue by invexed** [HowardHinnant/date#459]
A -Wnoexcept warning is triggered when compiling with GCC 9.1.0:

```cpp
date-master/src/tz.cpp: In function 'std::istream& date::detail::operator>>(std::istream&, date::detail::MonthDayTime&)':
date-master/src/tz.cpp:965:22: warning: noexcept-expression evaluates to 'false' because of a call to 'constexpr std::chrono::duration<_Rep, _Period>::duration(const _Rep2&) [with _Rep2 = int; <template-parameter-2-2> = void; _Rep = long int; _Period = std::ratio<3600>]' [-Wnoexcept]
  965 |     x = MonthDayTime{};
      |                      ^
```
\
For now I have suppressed it by adding an appropriate noexcept-specification to the default constructor on line 104 of tz_private.h:
```cpp
MonthDayTime() noexcept(false) = default;
```

**Comment by HowardHinnant** [comment]
I'm not positive (I don't have gcc 9.1), but another way to suppress this may be to change just above in tz_private.h:
```
    std::chrono::hours           h_{0};
    std::chrono::minutes         m_{0};
    std::chrono::seconds         s_{0};
```
to:
```
    std::chrono::hours           h_{};
    std::chrono::minutes         m_{};
    std::chrono::seconds         s_{};
```

**Comment by invexed** [comment]
Good catch! The change you suggested does indeed suppress the warning. Any idea why? `std::chrono::duration`'s default constructor isn't `noexcept` either according to cppreference.

**Comment by HowardHinnant** [comment]
The `duration` default constructor is implicitly `noexcept` if the `Rep`'s default constructor is `noexcept`, which it is when the `Rep` is integral.

This would be a good change.

**Comment by invexed** [comment]
Merged: 1f5c192
